File: user-service/src/main/java/com/petstore/userservice/domain/service/UserService.java
package com.petstore.userservice.domain.service;

import java.util.UUID;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.petstore.userservice.api.dto.request.RegisterRequest;
import com.petstore.userservice.api.dto.request.UpdateUserRequest;
import com.petstore.userservice.api.dto.response.UserStatisticsResponse;
import com.petstore.userservice.domain.model.User;

public interface UserService {
    // User operations
    User createUser(String keycloakId, RegisterRequest request);
    User updateUser(String keycloakId, UpdateUserRequest request);
    User getProfile(String keycloakId);
    
    // Admin operations
    Page<User> getAllUsers(Pageable pageable);
    User getUserById(UUID userId);
    Page<User> searchUsers(String keyword, Pageable pageable);
    User updateUserById(UUID userId, UpdateUserRequest request);
    void deleteUser(UUID userId);
    UserStatisticsResponse getUserStatistics();
}

File: frontend-admin/src/services/userService.js
// src/services/userService.js
import api from './api';

export const userService = {
  // Get all users (admin only)
  getAllUsers: async (page = 0, size = 20, sort = 'createdAt,desc') => {
    const response = await api.get('/api/public/users/all', {
      params: { page, size, sort },
    });
    return response.data;
  },

  // Get user by ID (admin only)
  getUserById: async (userId) => {
    const response = await api.get(`/api/public/users/${userId}`);
    return response.data;
  },

  // Search users (admin only)
  searchUsers: async (keyword, page = 0, size = 20) => {
    const response = await api.get('/api/public/users/search', {
      params: { keyword, page, size },
    });
    return response.data;
  },

  // Create user (admin only)
  createUser: async (userData) => {
    const response = await api.post(`/api/auth/register`, userData);
    return response.data;
  },

  // Update user (admin only)
  updateUser: async (userId, userData) => {
    const response = await api.put(`/api/public/users/${userId}`, userData);
    return response.data;
  },

  // Delete user (admin only)
  deleteUser: async (userId) => {
    const response = await api.delete(`/api/public/users/${userId}`);
    return response.data;
  },

  // Get user statistics (admin only)
  getUserStatistics: async () => {
    const response = await api.get('/api/public/users/summary');
    return response.data;
  },

  // Get current user's profile (authenticated user)
  getProfile: async () => {
    const response = await api.get('/api/me/profile');
    return response.data;
  },

  // Get user's shipping addresses
  getUserShippingAddresses: async (userId) => {
    const response = await api.get(`/api/public/users/${userId}/shipping-info`);
    return response.data;
  },

  // Get user's favorites
  getUserFavorites: async (userId) => {
    const response = await api.get(`/api/public/users/${userId}/favorites`);
    return response.data;
  },
};

